### PR TITLE
Issue #6421 - Nima media support

### PR DESCRIPTION
**Pull request by Tomas-Kraus**
Fixes issue #6421

### CODE REVIEW

File: nima/http/media/jsonp/src/main/java/io/helidon/nima/http/media/jsonp/JsonpMediaSupportProvider.java [hunk 1]
```diff
@@ -16,127 +16,38 @@
...
 
     @Override
-    public <T> WriterResponse<T> writer(GenericType<T> type, WritableHeaders<?> requestHeaders) {
-        // if the type is JsonStructure, we support it and do not care about content type
-        // you provide json, you get json...
-        if (isSupportedType(type)) {
-            return new WriterResponse<>(SupportLevel.SUPPORTED, this::writer);
-        }
-        return WriterResponse.unsupported();
+    public double weight() {
```
Comment (tomas-langer) [inline nima/http/media/jsonp/src/main/java/io/helidon/nima/http/media/jsonp/JsonpMediaSupportProvider.java:48]: JSON-P has to have higher than both JSON-B and Jackson, but it does not need to be low, as it supports very specific types (`JsonObject`, `JsonArray` etc.)

File: nima/webserver/webserver/src/test/java/io/helidon/nima/webserver/WebServerConfigTest.java [hunk 2]
```diff
@@ -148,7 +147,7 @@ void testMediaSupportManualConfigJson() throws IOException {
                 .config(config.get("server"))
                 .mediaContext(MediaContext.builder()
                         .discoverServices(false)
-                        .addMediaSupportProvider(new JsonpMediaSupportProvider())
+                        .addMediaSupport(new JsonpSupport())
```
Comment (tomas-langer) [inline nima/webserver/webserver/src/test/java/io/helidon/nima/webserver/WebServerConfigTest.java:150]: All MediaSupport implementation must have a private/package local/protected constructor only (whatever is the lowest one we need to use).
So you cannot create an instance like this (this should fail compilation). 
Please fix it for all supports, and use `create()` or `create(Config)` methods instead

**Review by tomas-langer — CHANGES_REQUESTED** [review]

File: common/config/pom.xml [hunk 3]
```diff
@@ -28,4 +28,39 @@
     <artifactId>helidon-common-config</artifactId>
     <name>Helidon Common Config</name>
 
+    <properties>
```
[the remark this answers is not in the document]
Comment (Tomas-Kraus) [reply common/config/pom.xml]: Removed

File: common/config/pom.xml [hunk 4]
```diff
@@ -28,4 +28,39 @@
...
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.hamcrest</groupId>
+            <artifactId>hamcrest-all</artifactId>
+            <scope>test</scope>
+        </dependency>
+    </dependencies>
+
+    <build>
```
[the remark this answers is not in the document]
Comment (Tomas-Kraus) [reply common/config/pom.xml]: Removed.

File: nima/webserver/webserver/src/test/java/io/helidon/nima/webserver/WebServerConfigTest.java (hunk 2 above)
Comment (Tomas-Kraus) [reply nima/webserver/webserver/src/test/java/io/helidon/nima/webserver/WebServerConfigTest.java]: All except String support have private constructors + create factory method.
StringSupport class is being extended in tests so constructor has protected constructor.

File: nima/http/media/jsonp/src/main/java/io/helidon/nima/http/media/jsonp/JsonpMediaSupportProvider.java (hunk 1 above)
Comment (Tomas-Kraus) [reply nima/http/media/jsonp/src/main/java/io/helidon/nima/http/media/jsonp/JsonpMediaSupportProvider.java:48]: OK. As I understand all the notes here and in the code, weights shall be JSON-P > Jackson > JsonB
and JSON-P shall be high, while Jackson and JsonB very low.
It's set to:

- JsonbMediaSupportProvider - 10
- JacksonMediaSupportProvider - 15
- JsonpMediaSupportProvider - 1000

**Review by tomas-langer — APPROVED** [review]

**Pull request merged**